web/src/stream_ui_updates.ts

```diff
@@ -218,6 +218,27 @@ export function update_settings_button_for_sub(sub: StreamSubscription): void {
     }
 }
 
+export function update_settings_button_for_archive_and_unarchive(sub: StreamSubscription): void {
+    if (!hash_parser.is_editing_stream(sub.stream_id)) {
+        return;
+    }
+
+    // This is for the Archive/Unarchive button in the right panel.
+    const $archive_button = $(
+        `.stream_settings_header[data-stream-id='${CSS.escape(sub.stream_id.toString())}'] .deactivate`,
+    );
+    const $unarchive_button = $(
+        `.stream_settings_header[data-stream-id='${CSS.escape(sub.stream_id.toString())}'] .reactivate`,
+    );
+    if (sub.is_archived) {
```

Comment:
Missed this before, but the archive and unarchive buttons should also be hidden if user does not have the required permission.